File: php/features.php
<?php
declare(strict_types=1);

// RonSwansonQuotes SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RonSwansonQuotesFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RonSwansonQuotesBaseFeature();
            case "test":
                return new RonSwansonQuotesTestFeature();
            default:
                return new RonSwansonQuotesBaseFeature();
        }
    }
}

File: py/features.py
# RonSwansonQuotes SDK feature factory

from feature.base_feature import RonSwansonQuotesBaseFeature
from feature.test_feature import RonSwansonQuotesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RonSwansonQuotesBaseFeature(),
        "test": lambda: RonSwansonQuotesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
